presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -343,18 +343,18 @@ public boolean isFinished()
         return finished;
     }
 
+    // This operator can receive input from the upstream operator only when all of following conditions are true:
+    // - 1. It has not received finish() signal (more input to come).
+    // - 2. The output pages have been flushed to the downstream operator.
+    // - 3. Aggregation has processed all the input.
```

Comment:
For partial aggregation, it actually will start aggregation processing when memory limit is hit, so it hasn't processed all the input

`// - 3. Aggregation builder has not been triggered or has finished processing.`